robosat/osm/road.py

```diff
@@ -0,0 +1,143 @@
+import sys
+
+import math
+import osmium
+import geojson
+import shapely.geometry
+
+from robosat.osm.core import is_polygon
+
+
+class RoadHandler(osmium.SimpleHandler):
+    """Extracts road polygon features (visible in satellite imagery) from the map.
+    """
+
+    #  highway=* to discard because these features are not vislible in satellite imagery
+    @staticmethod
+    def road_filter():
+        return {
+            "motorway",
+            "trunk",
+            "primary",
+            "secondary",
+            "tertiary",
+            "unclassified",
+            "residential",
+            "service",
+            "motorway_link",
+            "trunk_link",
+            "primary_link",
+            "secondary_link",
+            "tertiary_link"
+        }
+
+    def __init__(self):
+        super().__init__()
+        self.features = []
+
+    def way(self, w):
+        if "highway" not in w.tags:
+            return
+
+        if w.tags["highway"] not in self.road_filter():
+            return
+
+        left_hard_shoulder_width = 0.0
+        lane_width = 3.0
+        lanes = 1
+        right_hard_shoulder_width = 0.0
```

Comment:
Below we basically hard-code a lookup from highway tag to these values. What about refactoring it to a dict we can use for the lookup? Such as

```python
highway_attributes = {
  'motorway': { 'lanes': 4, ... },
}
```

and then simply index into that with the highway type.

Reply:
I agree with it. It would be clear to use a dict.